symplyphysics/laws/thermodynamics/entropy_from_statistical_weight.py

```diff
@@ -16,9 +24,186 @@
+
+_first_probability_diff_expr = solve(
+    _constant_condition,
+    _first_probability.diff(_time)
+)[0]
+
+_additive_property_diff = Eq(
+    _additive_property.lhs.diff(_time),
+    _additive_property.rhs.diff(_time)
+).subs(
+    _first_probability.diff(_time),
+    _first_probability_diff_expr
+)
+
+_first_probability_expr = solve(
+    _additive_property_diff,
+    _first_probability
+)[0]
+
+_differential_property = Eq(
+    _first_probability * _entropy(_first_probability).diff(_first_probability),
+    _first_probability_expr * _entropy(_first_probability).diff(_first_probability),
+)
+
+assert expr_equals(_differential_property.lhs.diff(_second_probability), 0)
+assert expr_equals(_differential_property.rhs.diff(_first_probability), 0)
+
+# Since both `P_1` and `P_2` can take arbitrary values in the range of their definition,
+# the function described by `_differential_property` must be constant for all values of `P`.
+# Since `f(P)` is universal for all bodies, the constant must be universal as well
+
+_probability = symbols("probability", positive=True)
+_equation_constant = symbols("equation_constant", real=True)
+
+_entropy_eqn = Eq(
+    _differential_property.lhs.subs(_first_probability, _probability),
+    _equation_constant,
+)
+
+_integration_constant = symbols("integration_constant", real=True)
+
+# Now we can integrate the above differential property and show that the integration constant is
+# actually zero.
+
+_entropy_via_probability = dsolve(
+    _entropy_eqn,
+    _entropy(_probability)
+).rhs.subs(
+    "C1",
+    _integration_constant,
+)
+
+_additive_property_updated = _additive_property.replace(
+    _entropy,
+    lambda probability: _entropy_via_probability.subs(_probability, probability)
+)
+
+_integration_constant_expr = solve(
+    _additive_property_updated,
+    _integration_constant,
+)[0]
+
+assert expr_equals(_integration_constant_expr, 0)
+
+_entropy_via_probability = _entropy_via_probability.subs(
+    _integration_constant,
+    _integration_constant_expr,
+)
+
+# Now, the question is to find the numeric value of the equation constant.
+# To do that, we can compare the entropy difference in two arbitrary states and the
+# logarithm of the probability ratio of these two states.
+
+# Let `V` be some volume containing `N` ideal gas particles and `V_0` be a part of it.
+# Then the probability of finding one molecule in `V_0` is `V_0 / V`, and the probability
+# of finding all `N` particles is `(V_0 / V)**N`
+
+_volume, _total_volume, _particle_count = symbols(
+    "volume total_volume particle_count",
+    positive=True,
+)
+
+_probability_via_volume = (_volume / _total_volume)**_particle_count
+
+_entropy_via_volume = _entropy_via_probability.subs(_probability, _probability_via_volume)
+
+# Let us assume that during a process the volume of the gas changes from `V_1` to `V_2` at
+# constant temperature. Then we can find the entropy of the two states via the above formula.
```

Comment:
Why temperature matters here? I think particle count should be constant instead.